accesshandler/pkg/providers/aws/sso/options.go

```diff
@@ -66,3 +80,25 @@ func (p *Provider) Options(ctx context.Context, arg string) ([]types.Option, err
 	return nil, &providers.InvalidArgumentError{Arg: arg}
 
 }
+
+func (p *Provider) checkPermissionSetIsTagged(ctx context.Context, permissionSetARN string) (bool, error) {
+	hasMore := true
+	var nextToken *string
+	for hasMore {
+		tags, err := p.client.ListTagsForResource(ctx, &ssoadmin.ListTagsForResourceInput{
+			InstanceArn: aws.String(p.instanceARN.Get()),
+			ResourceArn: aws.String(permissionSetARN),
+		})
+		if err != nil {
+			return false, err
+		}
+		nextToken = tags.NextToken
+		hasMore = nextToken != nil
+		for _, tag := range tags.Tags {
+			if aws.ToString(tag.Key) == "commonfate.io/permission-set" {
```

Comment:
We can use a more generic tag to indicate the resource is managed/accessible by granted

```suggestion
			if aws.ToString(tag.Key) == "commonfate.io/managed-by-granted" {
```

Reply:
Related to this, for EKS provider granted also creates permission sets, we shoudl probably tag those too, but we don't want them to show up in the UI for the SSO provider.

maybe for these EKS ones we use `commonfate.io/created-by-granted` 

I also think that I should be updating the SSO provider validation check to ensure that only permissions sets with this tag can be granted as part of this PR

Reply:
@chrnorm I have added this, but I realised that the Validater interface is only used in the integration test, It's not clear to me what your original intention was for this.

I will leave it out of scope for this ticket